### DISCUSSION TITLE
Execution in a Cloud Function

### DISCUSSION THREAD

**Discussion by gilesbutler** [ElMassimo/iles#121] [Q&A] [answered]
Hi @ElMassimo,

Our setup requires running `iles build` from within a Google cloud function/

We're getting the following error when we execute the command

```
failed to load config from /workspace/iles.config.ts
build error: Error: EROFS: read-only file system, open '/workspace/iles.config.ts.js'
at Object.openSync (node:fs:585:3)
at Object.writeFileSync (node:fs:2153:35)
at loadConfigFromFile (/workspace/node_modules/vite/dist/node/chunks/dep-88bd5805.js:61820:29)
at processTicksAndRejections (node:internal/process/task_queues:96:5)
at async loadUserConfigFile (/workspace/node_modules/iles/dist/node/chunk-EBSYBTDU.js:108:35)
at async resolveUserConfig (/workspace/node_modules/iles/dist/node/chunk-EBSYBTDU.js:82:14)
at async resolveConfig (/workspace/node_modules/iles/dist/node/chunk-EBSYBTDU.js:65:21)
at async build (/workspace/node_modules/iles/dist/node/chunk-YTVPIDBU.js:33:21)
```

My assumption is this is because Iles is trying to convert `iles.config.ts` to `iles.config.ts.js` when it builds, or possibly because it's trying to create `.iles-ssg-temp` both of which aren't allowed as you're not allowed to write to the functions folder in the cloud function. You can however write to `/tmp` which is where we're pointing the config's `outDir` to. Would love to get your thoughts on a workaround for this.

Thanks,
Giles

**Comment by ElMassimo** [comment, marked as answer]
Hi @gilesbutler!

Regarding `.iles-ssg-temp`, you can configure that using [`tempDir`](https://github.com/ElMassimo/iles/blob/main/packages/iles/src/node/config.ts#L214).

`loadConfigFromFile` is exposed by Vite, and it does [create a temporary file](https://github.com/vitejs/vite/blob/main/packages/vite/src/node/config.ts#L809) when transpiling TypeScript.

Although it's technically possible to support [`iles.config.js`](https://github.com/ElMassimo/iles/blob/main/packages/iles/src/node/config.ts#L103), the [resolution process is very complex](https://github.com/vitejs/vite/blob/b5c370941bb36bdb420433ca16cea9c2402b9810/packages/vite/src/node/config.ts#L760-L788).

Even if tackling this problem, not being able to write to the filesystem is a limitation that can cause other problems inside Vite.

I'd recommend you experiment by [forking iles and adding support for all extensions for `iles.config`](https://github.com/vitejs/vite/blob/b5c370941bb36bdb420433ca16cea9c2402b9810/packages/vite/src/node/config.ts#L760-L788), use `iles.config.js` or `iles.config.mjs`, and then see if any other problems come up.

If it turns out everything runs as expected, we can bring the file detection into iles.

**Reply by gilesbutler** [reply]
Thanks for the detailed response @ElMassimo 👍

> Even if tackling this problem, not being able to write to the filesystem is a limitation that can cause other problems inside Vite.

I think we'll take a look at bringing Vite and Iles out of the cloud function and running them somewhere else instead. It did feel a bit messy doing it that way anyway.
